src/gui/progressbardelegate.cpp

```diff
@@ -28,47 +28,41 @@
 
 #include "progressbardelegate.h"
 
-#include <QApplication>
-#include <QModelIndex>
 #include <QPainter>
+#include <QPalette>
 #include <QStyleOptionViewItem>
 
 #if (defined(Q_OS_WIN) || defined(Q_OS_MACOS))
 #include <QProxyStyle>
 #endif
 
-ProgressBarDelegate::ProgressBarDelegate(const int progressColumn, const int dataRole, QObject *parent)
-    : QStyledItemDelegate {parent}
-    , m_progressColumn {progressColumn}
-    , m_dataRole {dataRole}
+ProgressBarDelegate::ProgressBarDelegate()
 {
 #if (defined(Q_OS_WIN) || defined(Q_OS_MACOS))
-    m_dummyProgressBar.setStyle(new QProxyStyle {"fusion"});
+    QProxyStyle style {"fusion"};
+    m_dummyProgressBar.setStyle(&style);
```

Comment:
>there was a memory leak here 🙄 

You changed it with "dangling pointer", don't you?

Reply:
It is unclear how Qt intend to handle such pointer, so I made it a `static` variable.

Reply:
>I made it a `static` variable.

👎 
Just make it private class field.

Reply:
or if I am correct there should be QStyle::setParent

Reply:
>It is unclear how Qt intend to handle such pointer

Usually, if a function takes a parameter by pointer, then the caller is still responsible for its lifetime (unless other behavior is explicitly specified).

Reply:
>Just make it private class field.

There is another advantage using `static` variable: all progress bar will use the same QProxyStyle instance.

>QStyle::setParent

Seems I'm forced to use this.